docs/melange-for-x-developers.md

```diff
@@ -0,0 +1,633 @@
+the specific details have been discussed in [the OCaml
+forum](https://discuss.ocaml.org/t/ahrefs-is-now-built-with-melange/12107/3).
+
+### Source-based vs pre-built distribution
+
+While with ReScript every dependency can be downloaded with just npm, Melange
+projects will have to use opam and npm. This is a trade-off: on one hand, some
+Melange projects might need to include two package configuration files. But on
+the other hand they can benefit from opam’s source-based package distribution
+model for things like PPXs, linters, or any other OCaml tooling.
+
+By comparison, to consume any OCaml ecosystem tool in ReScript is more
+challenging. As npm is a package manager designed for an interpreted language,
+package distribution requires including prebuilt binaries, which makes
+maintenance harder for library authors, and might lead to issues for library
+users if their architecture is not included in the prebuilt set of binaries.
+
+### OCaml compiler version
+
+ReScript is compatible with the 4.06 version of the OCaml compiler, while
+Melange is compatible with the version 4.14 (as of May 2023).
+
+The (Melange roadmap)[todo-fix-me.md] includes a milestone to upgrade to the
+latest version of the compiler 5.0.
+
+### Editor integration
+
+Melange is fully compatible with the [OCaml platform editor
+tools](https://ocaml.org/docs/up-and-running#configuring-your-editor), which
+makes possible to work in projects that include OCaml and Melange code using the
+same editor configuration.
+
+ReScript has its [own set of editor
+plugins](https://rescript-lang.org/docs/manual/latest/editor-plugins).
+
+### Feature choice and alignment with OCaml
+
+ReScript goal is to model the language to bring it as close to JavaScript as
+possible. From the website [introduction
+section](https://rescript-lang.org/docs/manual/latest/introduction):
+
+> ReScript looks like JS, acts like JS, and compiles to the highest quality of
+> clean, readable and performant JS (...)
+
+New features added to ReScript might close its alignment with JavaScript, but
+some of these features can lead to greater divergence from OCaml. As Melange
+prioritizes compatibility with OCaml, it avoids incorporating those features
+that widen the gap between the two.
+
+Here is a non-exhaustive list of the features that ReScript has added and will
+not be supported in Melange:
+
+- The `async` / `await` syntax: similar functionality can be achieved in Melange
+  through the usage of [binding
+  operators](https://v2.ocaml.org/manual/bindingops.html) (introduced in OCaml
+  4.13).
+- Optional fields in records, like `type t = { x : int, @optional y : int }`.
+- Uncurried by default.
+
+The restriction above only applies to features that compromise compatibility
+with OCaml, but otherwise Melange can incorporate bugfixes or new functionality
+from ReScript.
+
+On the other hand, as Melange goal is to keep up with the version of the OCaml
```

Comment:
We aren't in a position to make this statement.

Reply:
I agree. What do you think about https://github.com/melange-re/documentation-site/pull/39/commits/ae675c534e65567f187eb83b77ad6145735c165e?